src/fromager/constraints.py

```diff
@@ -0,0 +1,54 @@
+import copy
+import logging
+import pathlib
+from io import TextIOWrapper
+
+from packaging.requirements import Requirement
+
+logger = logging.getLogger(__name__)
+
+
+class Constraints:
+    def __init__(self, data: dict[str, Requirement]):
+        self._data = data
+
+    def get_new_requirement(self, req: Requirement):
+        constraint = self._data.get(req.name)
+        new_req = copy.deepcopy(req)
+
+        if not constraint:
+            return new_req, None
+
+        # only allow "==" in constraints for now
+        if len(constraint.specifier) != 1 or not str(constraint.specifier).startswith(
+            "=="
+        ):
+            logger.debug(
```

Comment:
This should probably raise an exception instead of just logging. The exception will make it clear that the user is doing something unsupported, instead of hiding that and continuing the build operations.